pyproject.toml

```
@@ -146,25 +146,22 @@ ECM_Example = "pybamm.input.parameters.ecm.example_set:get_parameter_values"
 MSMR_Example = "pybamm.input.parameters.lithium_ion.MSMR_example_set:get_parameter_values"
 Chayambuka2022 = "pybamm.input.parameters.sodium_ion.Chayambuka2022:get_parameter_values"
 
-[tool.setuptools]
-include-package-data = true
+[tool.hatch.build]
+packages = ["src/pybamm"]
+include = [
+    "src/pybamm/**/*.txt",
+    "src/pybamm/**/*.md",
+    "src/pybamm/**/*.csv",
+    "src/pybamm/**/*.py",
+    "src/pybamm/CITATIONS.bib",
+    "src/pybamm/plotting/mplstyle",
```

`include-package-data` was used to include the files from [`MANIFEST.in`](https://github.com/pybamm-team/PyBaMM/blob/develop/MANIFEST.in) in the wheel file.

`hatchling` includes everything not ignored by `.gitignore` by default. Therefore, we should just include the required files -

```suggestion
[tool.hatch]
build.targets.sdist.include = [
  "src/pybamm",
  "CITATION.cff",
] 
```

We should not bundle the `CITATION.cff` file with the wheel (as it is being done now). We should also get rid off `MANIFEST.in`.

---

@vidipsingh This change is still needed